src/Trustless.ts

```diff
@@ -0,0 +1,56 @@
+import fetchPonyfill from 'fetch-ponyfill'
+
+import { CheckBase } from './CheckBase'
+import { HASH_TO_TEST, TRUSTLESS_RESPONSE_TYPES } from './constants'
+import type { GatewayNode } from './GatewayNode'
+
+import { Log } from './Log'
+
+const { fetch } = fetchPonyfill()
+
+const log = new Log('Trustless')
+
+class Trustless extends CheckBase implements Checkable {
+  _className = 'Trustless'
+  _tagName = 'div'
+  constructor (protected parent: GatewayNode) {
+    super(parent, 'div', 'Trustless')
+  }
+
+  async check (): Promise<void> {
+    const now = Date.now()
+    const gatewayAndHash = this.parent.gateway.replace(':hash', HASH_TO_TEST)
+    this.parent.tag.classList.add('trustless')
+    try {
+      const trustlessResponseTypesTests = await Promise.all(TRUSTLESS_RESPONSE_TYPES.map(
+        async (trustlessTypes): Promise<boolean> => {
+          const testUrl = `${gatewayAndHash}?format=${trustlessTypes}&now=${now}#x-ipfs-companion-no-redirect`
+          const response = await fetch(testUrl)
+          return Boolean(response.headers.get('Content-Type')?.includes(`application/vnd.ipld.${trustlessTypes}`))
+        }
+      ))
+
+      if (!trustlessResponseTypesTests.includes(false)) {
+        this.tag.win()
+      } else {
+        log.debug('The response type did not match the expected type')
```

Comment:
For better troubleshooting, we should understand what the expected type was in the error message. i.e. If a gateway owner fails a test and wants to figure out how they failed, do they need to support car, raw, or both?

probably worth addressing in another issue.